### PR TITLE
Break circular reference of Spring beans

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus/src/integrationTest/java/org/vividus/action/IntegrationElementAction.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.action;
+
+import org.vividus.ui.action.search.IElementAction;
+import org.vividus.ui.action.search.LocatorType;
+
+public class IntegrationElementAction implements IElementAction
```
Comment (valfirst) [inline vividus/src/integrationTest/java/org/vividus/action/IntegrationElementAction.java:22]: why is it needed?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus/src/integrationTest/java/org/vividus/action/IntegrationElementAction.java (hunk 1 above)
Comment (uarlouski) [reply vividus/src/integrationTest/java/org/vividus/action/IntegrationElementAction.java:22]: At least one IElementAction is required to satisfy @Inject Set<IElementAction>, btw just declaring set collection with IElementAction typing doesn't work

File: vividus-extension-selenium/src/main/resources/spring.xml [hunk 2]
```diff
@@ -9,6 +9,10 @@
         http://www.springframework.org/schema/util https://www.springframework.org/schema/util/spring-util.xsd"
         default-lazy-init="true" profile="web,mobile_app">
 
+    <context:component-scan base-package="org.vividus.bdd.converter.ui" >
```
Comment (ikalinin1) [inline vividus-extension-selenium/src/main/resources/spring.xml:12]: Why it is required?

Comment (uarlouski) [reply vividus-extension-selenium/src/main/resources/spring.xml]: resolved in other pr

File: vividus-extension-selenium/src/test/java/org/vividus/ui/action/search/ElementActionServiceTests.java [hunk 3]
```diff
@@ -20,22 +20,36 @@
...
-    ));
+    private ElementActionService service;
+
+    @BeforeEach
+    void init() throws IllegalArgumentException, IllegalAccessException
+    {
+        service = new ElementActionService();
+        Set<IElementAction> elementActions = Set.of(new TestElementFilter(), new TestElementSearch());
+        Field field = findFields(ElementActionService.class, f -> "elementActions".equals(f.getName()),
+                HierarchyTraversalMode.TOP_DOWN).get(0);
```
Comment (valfirst) [inline vividus-extension-selenium/src/test/java/org/vividus/ui/action/search/ElementActionServiceTests.java:48]: what about mockito `Spy`?

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**